ci/appveyor-cpp-setup.bat

```diff
@@ -55,6 +55,7 @@ mamba update -q -y -c conda-forge --all || exit /B
 @rem Create conda environment
 @rem
 
+set CONDA_DLL_SEARCH_MODIFICATION_ENABLE=1
```

Comment:
Can you add a short comment explaining why this line is needed?

Reply:
Yes, I can try - I don't really know why this solves it really 🤣 But it apparently does.
I will nevertheless try to add a meaningful comment 👍

Reply:
I have added this comment. Pls let me know if it is not clear ..

```
@rem Workaround for ARROW-17172
@rem Due to failing test_cython_api() import of an external module in the
@rem subprocess this env var is added for os.add_dll_directory to work as
@rem expected in the conda environment and for the import to succeed.
@rem See: https://github.com/ContinuumIO/anaconda-issues/issues/12475
```